### PR TITLE
Build + test fixes to prepare for 4.1 release

### PR DESCRIPTION
**Pull request by znation**
* Remove unused spacy dependency in scenario tests
* Bump coremltools dependency to 0.8
* Use assertAlmostEqual to get better failure msg in test_perplexity
* Get scenario tests ready for Python 3 execution
* Expect wheels on Linux for scenario and unit tests
* Make wheels on Linux
* Rename make_egg -> make_wheel
* Bump version number to 4.1a1
* Use VIRTUALENV env variable in run_scenario_tests

### CODE REVIEW

File: src/unity/python/dummy.c [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include <stdio.h>
```
Comment (TobyRoseman) [inline src/unity/python/dummy.c]: Why is this file needed?

Comment (znation) [reply src/unity/python/dummy.c]: Probably isn't. I'll remove.
